base/linalg/bunchkaufman.jl

```diff
@@ -69,13 +71,8 @@ The following functions are available for
 [^Bunch1977]: J R Bunch and L Kaufman, Some stable methods for calculating inertia and solving symmetric linear systems, Mathematics of Computation 31:137 (1977), 163-179. [url](http://www.ams.org/journals/mcom/1977-31-137/S0025-5718-1977-0428694-0/).
 
 """
-bkfact(A::StridedMatrix{<:BlasFloat}, uplo::Symbol=:U, symmetric::Bool=issymmetric(A),
-    rook::Bool=false) =
-        bkfact!(copy(A), uplo, symmetric, rook)
-bkfact(A::StridedMatrix{T}, uplo::Symbol=:U, symmetric::Bool=issymmetric(A),
-    rook::Bool=false) where {T} =
-        bkfact!(convert(Matrix{promote_type(Float32, typeof(sqrt(one(T))))}, A),
-                uplo, symmetric, rook)
+bkfact(A::AbstractMatrix{T}, rook::Bool=false) where {T} =
+    bkfact!(copy_oftype(A, typeof(sqrt(one(T)))), rook)
```

Comment:
No. They are unaffected since `sqrt(one(T)) isa Float64` for `T in (Rational{Int},Int)`. A `Matrix{Float16}` will now fail instead of being promoted. I think that is reasonable because people using `Float16` most likely prefer an error over silent promotion and I really don't think anybody was ever called `bkfact` with a `Matrix{Float16}` in serious code.